File: scsc-monolith/src/main/java/demo/scsc/api/productcatalog/ProductListQueryResponse.java
package demo.scsc.api.productcatalog;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record ProductListQueryResponse(
        List<ProductInfo> products
) {
    public record ProductInfo(
            UUID id,
            String name,
            String desc,
            BigDecimal price,
            String image
    ) {
    }
}

File: scsc-monolith/src/main/kotlin/demo/scsc/api/productcatalog/ProductListQueryResponse.kt
package demo.scsc.api.productcatalog

import java.math.BigDecimal
import java.util.*

data class ProductListQueryResponse(val products: List<ProductInfo>) {
    data class ProductInfo(val id: UUID, val name: String, val desc: String, val price: BigDecimal, val image: String)
}
